projects/adrv902x/src/app/headless.c

```diff
@@ -0,0 +1,639 @@
+/***************************************************************************//**
+ *   @file   headless.c
+ *   @brief  adrv902x main project file.
+ *   @author GMois (george.mois@analog.com)
+********************************************************************************
+ * Copyright 2023(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "xil_cache.h"
+
+#include "no_os_print_log.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_spi.h"
+
+#include "axi_adc_core.h"
+#include "axi_dac_core.h"
+#include "axi_adxcvr.h"
+#include "axi_dmac.h"
+
+#include "parameters.h"
+#include "xilinx_spi.h"
+#include "xilinx_gpio.h"
+
+#include "adrv9025.h"
+#include "axi_jesd204_rx.h"
+#include "axi_jesd204_tx.h"
+
+#include "ad9528.h"
+
+#include "app_config.h"
+
+#ifdef IIO_SUPPORT
```

Comment:
so we've kind of switched to the new way of writing projects with multiple examples and I think you should adopt that too here

I think what needs to happen is that the user can run `make EXAMPLE=DMA` or `make EXAMPLE=IIO`. If EXAMPLE is not provided at command line, default to DMA. Examples should be mutually exclusive.

The good thing here is that you don't have a lot of choice regarding platforms. ;)

Reply:
Done